src/copyGoodPixels.cc

```diff
@@ -102,14 +100,58 @@ namespace {
             for (; srcIter != srcEndIter; ++srcIter, ++destIter) {
                 if (isValid(srcIter)) {
                     *destIter = *srcIter;
-//                     typename ImageT::SinglePixel pix = *srcIter;
-//                     *destIter = pix;
                     ++numGoodPix;
                 }
             }
         }
         return numGoodPix;
     }
+
+    template <typename ImageT, typename isValidPixel>
+    int copyGoodPixelsImpl(
+        ImageT &destImage,                                  ///< [in,out] image to modify
+        ImageT const &srcImage,                             ///< image to copy
+        lsst::afw::image::MaskPixel const badPixelMask,      ///< bad pixel mask; may be ignored
```

Comment:
Under what circumstances may it be ignored? It seems like you're always checking it for every overlapping pixel. Or do you just mean it might be ignored if there's no overlap?